aws-rds-cfn-common/src/main/java/software/amazon/rds/common/logging/ProxyClientLogger.java

```diff
@@ -0,0 +1,92 @@
+package software.amazon.rds.common.logging;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import software.amazon.awssdk.awscore.AwsRequest;
+import software.amazon.awssdk.awscore.AwsResponse;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.core.pagination.sync.SdkIterable;
+import software.amazon.cloudformation.proxy.ProxyClient;
+
+public class ProxyClientLogger {
+
+    public static <ClientT> ProxyClient<ClientT> newProxy(final RequestLogger requestLogger,
+                                                          final ProxyClient<ClientT> proxyClient) {
+        return new ProxyClient<ClientT>() {
```

Comment:
I don't think there is a reason to create anonymous classes upon every single invocation of `newProxy`.

Another way to get this done is:

```java
public class LoggingProxyClient<T> implements ProxyClient<T> {
    public LoggingProxyClient(final ProxyClient<T> delegate) {
        //...
    }
}
```

Reply:
Done.